app/components/filters_component.rb

```diff
@@ -1,34 +1,57 @@
 class FiltersComponent < GovukComponent::Base
-  attr_accessor :filters, :form, :items, :options
+  attr_accessor :filters, :form, :options
 
   def self.variants
     %w[default]
   end
 
-  def initialize(filters:, form:, items:, options:, classes: [], html_attributes: {})
+  def initialize(form:, options:, filters: {}, classes: [], html_attributes: {})
     super(classes: classes, html_attributes: html_attributes)
 
     @filters = filters
     @form = form
-    @items = items
     @options = options
   end
 
-  def render?
-    filters.present?
-  end
+  renders_many :groups, lambda { |key:, component:|
+    tag.div(key, class: "filters-component__groups__group", data: { group: key }) do
+      tag.div(component)
+    end
+  }
 
-  def display_remove_buttons?
-    filters[:total_count].positive? && options[:remove_buttons]
+  renders_one :remove_buttons, "RemoveButtons"
+
+  class RemoveButtons < GovukComponent::Base
+    def initialize(classes: [], html_attributes: {})
+      super(classes: classes, html_attributes: html_attributes)
+    end
+
+    renders_many :groups, lambda { |selected:, key:, options:, value_method:, selected_method:, legend: false|
+      if selected&.any?
+        safe_join([
+          tag.h4(legend, class: "govuk-heading-s"),
+          tag.ul(class: "filters-component__remove-tags") do
+            options.collect do |option|
+              next unless selected&.include?(option.send(value_method))
```

Comment:
You know `selected` exists at this point.

Reply:
Can we also make this a `public_send`?

Reply:
regarding the selected thing - yes i realise it exists as there is an outer check and then this line checks wether the current group of filters has one of its items currently selected i.e wether to display a remove button for it. am i misunderstanding your comment @elliotcm ?

Reply:
> and the comment above about extracting this - yes i was thinking it would be nice to do something like that as this groups lamba function does look particuarly nice but i was unsure whats the best way as it contains a bit of logic and the public_send stuff. maybe we can spend 20mins refactoring in mob on monday if nothing better to do. also a little concious of unblocking @josephhull676 abtest at the same time if we dont and its not a deal breaker...

Yeah it's not a big deal.

Reply:
> regarding the selected thing - yes i realise it exists as there is an outer check and then this line checks wether the current group of filters has one of its items currently selected i.e wether to display a remove button for it. am i misunderstanding your comment @elliotcm ?

What I mean here is that the `&.` is unnecessary because you know `selected` isn't nil.